access.lua

```diff
@@ -269,6 +276,13 @@ if permission ~= nil and ngx.req.get_headers()["Authorization"] ~= nil then
             ngx.req.clear_header("Authorization")
         end
     end
+
+    -- Clean all Yunohost Authentication Headers
+    for k, v in pairs(ngx.req.get_headers()) do
+        if string.starts(k, "YNH_") then
+            ngx.req.clear_header(k)
+        end
+    end
```

Comment:
Isn't this block supposed to happen no matter what and not just if an Authorization header is provided ?

Reply:
The idea there was juste to ensure that there are no way from the client to do user spoofing for all header `YNH_*`. So we remove all header `YNH_*` except the one which come from ssowat.

Reply:
i understand what this block does, what i'm pointing out is the fact that it's ran only if there's an Authorization header set, in which case the attacker could still spoof the YNH_USER header by not providining an Authorization header

Reply:
Yes, I understand now. I think in this case, it make more sense to move it out of the if. So we remove in any case the header and then, set it again if the user if authenticated.